src/api/vdom.js

```diff
@@ -133,36 +152,62 @@ function resolveTagName(tname) {
 function wrapIdomFunc(func, tnameFuncHandler = () => {}) {
   return function wrap(...args) {
     args[0] = resolveTagName(args[0]);
+    stackCurrentHelper = null;
     if (typeof args[0] === 'function') {
       // If we've encountered a function, handle it according to the type of
       // function that is being wrapped.
+      stackCurrentHelper = args[0];
       return tnameFuncHandler(...args);
     } else if (stackChren.length) {
       // We pass the wrap() function in here so that when it's called as
       // children, it will queue up for the next stack, if there is one.
       stackChren[stackChren.length - 1].push([wrap, args]);
     } else {
-      // If there is no stack left, we call Incremental DOM directly.
+      const isElementClosing = func === elementClose;
+      const isElementOpening = func === elementOpen;
+
+      // If we're skipping the tree, we must skip everything except for the
+      // closing of the element that originally started the skipping.
+      if (skipCurrentTree && !isElementClosing && !currentElement()[$skipCurrentElement]) {
+        return;
+      }
+
       const elem = func(...args);
 
-      // If we're in elementClose, try calling the ref.
-      if (func === elementClose) {
+      if (isElementOpening && elem[$skipCurrentElement]) {
+        skipCurrentTree = true;
+      } else if (isElementClosing) {
         const eref = elem[$ref];
+
+        // We delete so that it isn't called again for the same element. If the
+        // ref changes, or the element changes, this will be defined again.
+        delete elem[$ref];
+
+        // Execute the saved ref after esuring we've cleand up after it.
         if (typeof eref === 'function') {
           eref(elem);
```

Comment:
nitpick, but the function name is a bit confusing without context. Even `executeRef` might be easier? (shrug)